apps/autoscheduler/get_host_target.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+// Print the host target to stdout.
+// Any extra arguments are assumed to be features that should be stripped from
+// the target (as a convenience for use in Makefiles, where string manipulation
+// can be painful).
+int main(int argc, char **argv) {
+    Target t = get_host_target();
+    for (int i = 1; i < argc; ++i) {
+        auto f = Target::feature_from_name(argv[i]);
+        if (f == Target::FeatureEnd) {
+            fprintf(stderr, "Unknown feature: %s\n", argv[i]);
+            exit(1);
+        }
+        t = t.without_feature(f);
+    }
+    printf("%s", t.to_string().c_str());
+    return 0;
+}
+
+
+
```

Comment:
I've been doing: 
```
any.generator -o foo -r bar target=pants 2>&1 | grep "the host target is" | cut -d: -f2
```

Perhaps any generator should be able to print the host target? Not worth changing for this PR though.

Reply:
Sniffing the host target by grepping an error message is expedient, but perhaps suboptimal as a long-term solution :-)

Reply:
Oh absolutely, but I wondered if generator binaries should serve as the utilities for these sorts of queries (much as they can generate runtimes).